tests/third_party/cupy/linalg_tests/test_decomposition.py

```diff
@@ -0,0 +1,248 @@
+        result_cpu = numpy.linalg.svd(a_cpu, full_matrices=self.full_matrices)
+        result_gpu = cupy.linalg.svd(a_gpu, full_matrices=self.full_matrices)
+        # Check if the input matrix is not broken
+        testing.assert_allclose(a_gpu, a_cpu)
+
+        assert len(result_gpu) == 3
+        for i in range(3):
+            assert result_gpu[i].shape == result_cpu[i].shape
+            if has_support_aspect64():
+                assert result_gpu[i].dtype == result_cpu[i].dtype
+            else:
+                assert result_gpu[i].dtype.kind == result_cpu[i].dtype.kind
+        u_cpu, s_cpu, vh_cpu = result_cpu
+        u_gpu, s_gpu, vh_gpu = result_gpu
+        testing.assert_allclose(s_gpu, s_cpu, rtol=1e-5, atol=1e-4)
+
+        # reconstruct the matrix
+        k = s_cpu.shape[-1]
+
+        # dpnp.dot/matmul does not support complex type and unstable on cpu
+        # TODO: remove it when dpnp.dot/matmul is updated
+        u_gpu = cupy.asnumpy(u_gpu)
+        vh_gpu = cupy.asnumpy(vh_gpu)
+        s_gpu = cupy.asnumpy(s_gpu)
+        xp = numpy
+
+        if len(shape) == 2:
+            if self.full_matrices:
+                a_gpu_usv = xp.dot(u_gpu[:, :k] * s_gpu, vh_gpu[:k, :])
+            else:
+                a_gpu_usv = xp.dot(u_gpu * s_gpu, vh_gpu)
+        else:
+            if self.full_matrices:
+                a_gpu_usv = xp.matmul(
+                    u_gpu[..., :k] * s_gpu[..., None, :], vh_gpu[..., :k, :]
+                )
+            else:
+                a_gpu_usv = xp.matmul(u_gpu * s_gpu[..., None, :], vh_gpu)
+        testing.assert_allclose(a_gpu, a_gpu_usv, rtol=1e-4, atol=1e-4)
+
+        # assert unitary
+        u_len = u_gpu.shape[-1]
+        vh_len = vh_gpu.shape[-2]
+        testing.assert_allclose(
+            xp.matmul(u_gpu.swapaxes(-1, -2).conj(), u_gpu),
+            stacked_identity(xp, shape[:-2], u_len, dtype),
+            atol=1e-4,
+        )
+        testing.assert_allclose(
+            xp.matmul(vh_gpu, vh_gpu.swapaxes(-1, -2).conj()),
+            stacked_identity(xp, shape[:-2], vh_len, dtype),
+            atol=1e-4,
+        )
+
+    @testing.for_dtypes(
+        [
+            numpy.int32,
+            numpy.int64,
+            numpy.uint32,
+            numpy.uint64,
+            numpy.float32,
+            numpy.float64,
+            numpy.complex64,
+            numpy.complex128,
+        ]
+    )
+    @testing.numpy_cupy_allclose(
+        rtol=1e-5, atol=1e-4, type_check=has_support_aspect64()
+    )
+    def check_singular(self, shape, xp, dtype):
+        array = testing.shaped_random(shape, xp, dtype=dtype, seed=self.seed)
+        a = xp.asarray(array, dtype=dtype)
+        a_copy = a.copy()
+        result = xp.linalg.svd(
+            a, full_matrices=self.full_matrices, compute_uv=False
+        )
+        # Check if the input matrix is not broken
+        assert (a == a_copy).all()
+        return result
+
+    # @condition.repeat(3, 10)
```

Comment:
we can uncomment that by updating `tests/third_party/cupy/testing/_condition.py` with:
```python
def repeat(times, intensive_times=None):
    ...
    casual_test = bool(int(os.environ.get("CUPY_TEST_CASUAL", "1")))
    ...
```
it will reduce the repeat number to 3